src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml

```diff
@@ -236,92 +236,145 @@ let main_fixed_lookup_tables () =
 
 (* Parameters *)
 (* nb of fixed lookup tables *)
-let fixed_lt_n = 1 + Random.State.int state 10
+let max_fixed_lt_n = 1 + Random.State.int state 10
 
 (* number of fixed lookups *)
-let fixed_lt_queries_n = 1 + Random.State.int state 10
+let fixed_lt_queries_n = 1 + Random.State.int state 100
 
 (* fixed lookup tables data *)
 let fixed_lt_data =
-  Array.init fixed_lt_n ~f:(fun i ->
-      let table_id = 1 + i in
-      let table_size = 1 + Random.State.int state 100 in
-      let indexes = Array.init table_size ~f:Field.Constant.of_int in
+  (* generate some random unique table ids *)
+  let fixed_table_ids =
+    Int.Set.to_array
+      (Int.Set.of_list
+         (List.init max_fixed_lt_n ~f:(fun _ ->
+              1 + Random.State.int state (max_fixed_lt_n * 4) ) ) )
+  in
+  Array.map fixed_table_ids ~f:(fun table_id ->
+      let max_table_size = 1 + Random.State.int state 100 in
+      let indices =
+        Int.Set.to_array
+          (Int.Set.of_list
+             (List.init max_table_size ~f:(fun _ ->
+                  1 + Random.State.int state (max_table_size * 4) ) ) )
+      in
+      let table_size = Array.length indices in
       let values =
-        Array.init table_size ~f:(fun _ ->
-            Field.Constant.of_int (Random.State.int state 100_000_000) )
+        Array.init table_size ~f:(fun _ -> Random.State.int state 100_000_000)
       in
-      (table_id, indexes, values) )
+      (table_id, indices, values) )
 
 (* lookup queries; selected random rows from f_lt_data *)
 let lookups =
   Array.init fixed_lt_queries_n ~f:(fun _ ->
-      let table_ix = Random.State.int state fixed_lt_n in
-      let table_id, indexes, values = fixed_lt_data.(table_ix) in
-      let table_size = Array.length indexes in
+      let table_id, indices, values =
+        fixed_lt_data.(Random.State.int state (Array.length fixed_lt_data))
+      in
+      let table_size = Array.length indices in
       let idx1 = Random.State.int state table_size in
       let idx2 = Random.State.int state table_size in
       let idx3 = Random.State.int state table_size in
       ( table_id
-      , (idx1, values.(idx1))
-      , (idx2, values.(idx2))
-      , (idx3, values.(idx3)) ) )
+      , (indices.(idx1), values.(idx1))
+      , (indices.(idx2), values.(idx2))
+      , (indices.(idx3), values.(idx3)) ) )
 
 let main_fixed_lookup_tables_multiple_tables_multiple_lookups () =
-  Array.iter fixed_lt_data ~f:(fun (table_id, indexes, values) ->
+  Array.iter fixed_lt_data ~f:(fun (table_id, indices, values) ->
```

Comment:
Nit: correct, but not common usage in code

Reply:
Will revert to `indexes` then + remove that comment.